Library/Homebrew/utils.rb

```diff
@@ -82,16 +82,19 @@ def odeprecated(method, replacement = nil, disable: false, disable_on: nil, call
   # - Location of caller of deprecated method (if all else fails).
   backtrace = caller
   tap_message = nil
+
+  # Don't throw deprecations at all for cached or .brew formulae.
+  return if backtrace.any? do |line|
+    line.include?(HOMEBREW_CACHE) || line.include?("/.brew/")
+  end
```

Comment:
Alternative that I find cleaner but don't feel too strongly on:

```ruby
return if backtrace.grep(/#{Regexp.escape(HOMEBREW_CACHE)}/)
return if backtrace.grep(%r{/.brew/})
```

Reply:
@alyssais Good call, changed (to use a string version)

Reply:
@alyssais Actually reverted back because neither string nor regex version of using `grep` works unfortunately.

Reply:
I’m not sure that works? Grep uses case equality (`===`), and afaik `”foo” === “f”` is false.

You could do it with two `any?`/`include?` lines, which would avoid both reflexes and the multi line postfix `if` (which is what currently sticks out to me).

Reply:
`”f === foo”`, even

Reply:
It doesn't work with the regexes either unfortunately. The current version works so I don't think it's worth fiddling with it further.

> You could do it with two any?/include? lines, which would avoid both reflexes and the multi line postfix if (which is what currently sticks out to me).

Unless it becomes very cluttered I think iterating once is better than twice.